var/spack/repos/builtin/packages/mctc-lib/package.py

```diff
@@ -27,6 +27,5 @@ class MctcLib(MesonPackage):
 
     def meson_args(self):
         return [
-            "--wrap-mode=nodownload",
```

Comment:
I only removed it because MesonPackage now applies it by default, so I don't have to enforce it on per package basis anymore.

Reply:
This is a pretty useful flag, but you might want to set it in the default meson args from MesonPackage instead.

Basically, it's common for projects to vendor code for... various reasons. System integrators, such as linux distros, homebrew, spack, etc. rarely like this, and prefer using dependency packages instead.

Meson tries to solve this by making it *really* easy to use vendored code. So easy, in fact, that no one would dream of doing it in a manner other than the officially approved one. :D And that manner is, doing standard system dependency lookup plus an ini file defining an url to vendor it with, and what dependencies it contains. Then, Meson can guarantee via a global policy how to handle that.

- by default, it's:
  - searched as a system dependency (spack depends_on, in this case)
  - if not found, it downloads and vendors a copy
- if `--wrap-mode=nodownload` is used, then it will never download a copy, and prefer to error out if it must, but if one is already downloaded, it's eligible for use
- if `--wrap-mode=nofallback` is used, vendoring will NEVER be done, missing system dependencies are always errors
- if `--wrap-mode=forcefallback` is used, vendoring is forced, and system dependencies are totally ignored when vendoring is possible


Either nodownload or nofallback are reasonable defaults for spack to use.

Reply:
Ah yes, just saw the new reply :D indeed, it's the default since 258edf7dac07710ea6c50316dc53e5154d9cb3b7 (which makes sense).